### PR TITLE
trust_env to the aiohttp.ClientSession

### PR DESCRIPTION
**Pull request by Temirlan-qa**
### Description
Passed `trust_env=True` to the `aiohttp.ClientSession `

Issues Resolved
closes [#368](https://github.com/opensearch-project/opensearch-py/issues/368)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Please sign DCO (amend the commit with `-s`) and add a unit test?

**Comment by margulanz** [comment]
It was already done in this pull request https://github.com/opensearch-project/opensearch-py/pull/370

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,6 +68,11 @@ jobs:
       - name: Run Tests
         run: |
           python setup.py test
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: ./junit/opensearch-py-codecov.xml
```
Comment (dblock) [inline .github/workflows/ci.yml:75]: These changes here seem unrelated. Make a separate PR to fix codecov?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Added
 - Added async support for helpers that are merged from opensearch-dsl-py ([#329](https://github.com/opensearch-project/opensearch-py/pull/329))
 ### Changed
+- Upgrading pytest-asyncio to latest version - 0.21.0 ([#339](https://github.com/opensearch-project/opensearch-py/pull/339))
```
Comment (dblock) [inline CHANGELOG.md:8]: Unrelated as well.

**Review by dblock — CHANGES_REQUESTED** [review]
Split up unrelated changes and it's good to go.

File: test_opensearchpy/test_async/test_connection.py [hunk 3]
```diff
@@ -97,6 +97,27 @@ def test_opaque_id(self):
         con = AIOHttpConnection(opaque_id="app-1")
         assert con.headers["x-opaque-id"] == "app-1"
 
+    def test_trust_env(self):
```
Comment (dblock) [inline test_opensearchpy/test_async/test_connection.py:100]: Add a test for defaults if we don't have one yet, `test_async_opensearch_default_options`, and one that changes all the options `test_async_opensearch_options`.

File: test_opensearchpy/test_async/test_connection.py [hunk 4]
```diff
@@ -97,6 +97,27 @@ def test_opaque_id(self):
...
+        con = AIOHttpConnection(trust_env=True)
+        assert con.session.connector.trust_env
+
+    async def test_async_opensearch_trust_env(self):
+        # Create an instance of AsyncOpenSearch with trust_env set to True
```
Comment (dblock) [inline test_opensearchpy/test_async/test_connection.py:105]: To be nitpicky, the comments are incorrect, because the test actually sets other options too, which they don't mention. 

I think they are simply unnecessary as the test and the code is self-explanatory. Would just remove to avoid being out of sync with the code.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -34,20 +34,19 @@ jobs:
       - name: Build project for distribution
         run: |
           python3.7 -m build
-
   test-linux:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['2.7', '3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ["2.7", "3.7", "3.8", "3.9", "3.10", "3.11"]
```
Comment (anabeel011) [inline .github/workflows/ci.yml:41]: I think these changes are unrelated.

**Comment by saimedhi** [comment]
Hello, @Temirlan-qa. Please fix the CI tests that are failing. If you have too many merge conflicts, you can also close this pull request and create a PR on a new branch that is in sync with the upstream source. Include only the changes that are relevant to the issue.

**Comment by saimedhi** [comment]
Hello @Temirlan-qa, Closing this pull request as it appears to be merged in PR https://github.com/opensearch-project/opensearch-py/pull/438
